internal/controller/pkg/manager/reconciler.go

```diff
@@ -285,6 +287,20 @@ func (r *Reconciler) Reconcile(ctx context.Context, req reconcile.Request) (reco
 		"name", p.GetName(),
 	)
 
+	// Check the pause annotation and return if it has the value "true"
+	// after logging, publishing an event and updating the SYNC status condition
+	if meta.IsPaused(p) {
+		log.Debug("Reconciliation is paused via the pause annotation", "annotation", meta.AnnotationKeyReconciliationPaused, "value", "true")
+		r.record.Event(p, event.Normal(reasonPaused, "Reconciliation is paused via the pause annotation"))
+		p.SetConditions(xpv1.ReconcilePaused())
+		// If the pause annotation is removed, we will have a chance to reconcile again and resume
+		// and if status update fails, we will reconcile again to retry to update the status
+		return reconcile.Result{}, errors.Wrap(r.client.Status().Update(ctx, p), errUpdateStatus)
+	}
+	if c := p.GetCondition(xpv1.ReconcilePaused().Type); c.Reason == xpv1.ReconcilePaused().Reason {
```

Comment:
I think we don't need this. 

After noticing that this is different than [managed reconciler](https://github.com/crossplane/crossplane-runtime/blob/89e89660b5c6237f14be6563e7b1f104a9ac3589/pkg/reconciler/managed/reconciler.go#L758-L767), I found this discussion which explains why we don't need to clean the conditions up: https://github.com/crossplane/crossplane-runtime/pull/357#discussion_r984071756

Reply:
Ah ok, I missed that package types not having a `Synced` condition. I am wondering if introducing a `Synced` annotation could make sense, but something to consider later.